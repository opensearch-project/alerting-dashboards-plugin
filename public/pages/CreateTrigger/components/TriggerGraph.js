/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { EuiSpacer } from '@elastic/eui';
import VisualGraph from '../../CreateMonitor/components/VisualGraph';
import TriggerExpressions from './TriggerExpressions';

const TriggerGraph = ({
  monitorValues,
  response,
  thresholdValue,
  thresholdEnum,
  fieldPath,
  flyoutMode,
}) => (
  <div style={flyoutMode ? {} : { padding: '0px 10px' }}>
    <TriggerExpressions
      thresholdValue={thresholdValue}
      thresholdEnum={thresholdEnum}
      keyFieldName={`${fieldPath}thresholdEnum`}
      valueFieldName={`${fieldPath}thresholdValue`}
      label="Trigger condition"
      flyoutMode={flyoutMode}
    />
    {!flyoutMode && (
      <>
        <EuiSpacer size="m" />
        <VisualGraph
          annotation
          values={monitorValues}
          thresholdValue={thresholdValue}
          response={response}
        />
      </>
    )}
  </div>
);

export default TriggerGraph;
