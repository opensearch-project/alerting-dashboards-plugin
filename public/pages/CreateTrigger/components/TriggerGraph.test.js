/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow } from 'enzyme';
import TriggerGraph from './TriggerGraph';

describe('TriggerGraph', () => {
  test('renders', () => {
    const wrapper = shallow(<TriggerGraph />);
    expect(wrapper).toMatchSnapshot();
  });
});
