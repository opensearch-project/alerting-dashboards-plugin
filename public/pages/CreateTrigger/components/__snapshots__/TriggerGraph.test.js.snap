// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriggerGraph renders 1`] = `
<div
  style={
    Object {
      "padding": "0px 10px",
    }
  }
>
  <TriggerExpressions
    keyFieldName="undefinedthresholdEnum"
    label="Trigger condition"
    valueFieldName="undefinedthresholdValue"
  />
  <EuiSpacer
    size="m"
  />
  <VisualGraph
    annotation={true}
  />
</div>
`;
